src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,242 @@
+}
+
+export enum TabSize {
+    Small = "Small",
+    Large = "Large",
+}
+
+export type TabsProps = {
+    paddingX?: TabsPaddingX;
+    size?: TabSize;
+    activeItemId: string;
+    children: ReactNode;
+    onChange?: (tabId: string) => void;
+};
+
+const paddingMap: Record<TabsPaddingX, string> = {
+    [TabsPaddingX.Small]: "tw-pl-s",
+    [TabsPaddingX.Medium]: "tw-pl-m",
+    [TabsPaddingX.Large]: "tw-pl-l",
+};
+
+export const Tabs: FC<TabsProps> = ({ paddingX, size, activeItemId, children, onChange }) => {
+    const [isOverflowing, setIsOverflowing] = useState(false);
+    const [isMenuOpened, setIsMenuOpened] = useState(false);
+    const layoutGroupId = useMemoizedId();
+
+    const tabs: TabItemProps[] =
+        Children.map(children, (child) => {
+            if (!isValidElement<TabItemProps>(child)) {
+                return null;
+            }
+            return child?.props;
+        }) ?? [];
+
+    const [tabIndexLimit, setTabIndexLimit] = useState(tabs.length);
+
+    const checkIfOverflowing = () => {
+        const tabNav = tabNavRef.current;
+        setIsOverflowing((tabNav && tabNav.scrollWidth > tabNav.clientWidth) ?? false);
+        const overFlowIndex = [];
+        if (tabNav) {
+            for (const tabItem of tabNav.children) {
+                const i = [...tabNav.children].indexOf(tabItem);
+                const navPosition = tabNav.getBoundingClientRect();
+                const tabPosition = tabItem.getBoundingClientRect();
+                if (tabPosition.right > navPosition.right) {
+                    overFlowIndex.push(i);
+                }
+            }
+
+            const overFlowIndexInAscOrder = [...overFlowIndex].sort((a, b) => a - b);
+            const newIndexLimit = overFlowIndex.length ? overFlowIndexInAscOrder[0] : tabs.length;
+            if (newIndexLimit < tabIndexLimit || newIndexLimit > tabIndexLimit) {
+                setTabIndexLimit(newIndexLimit);
+            }
+        }
+    };
+
+    const handleKeyboardTabChange = (e: KeyboardEvent<HTMLButtonElement>) => {
+        const nextTabs = tabs.filter((tab) => {
+            if (tab.id > activeItemId && !tab.disabled) {
+                return tab;
+            }
+        });
+
+        const previousTabs = tabs.filter((tab) => {
+            if (tab.id < activeItemId && !tab.disabled) {
+                return tab;
+            }
+        });
+
+        if ((e.key === "ArrowRight" || e.key === "ArrowDown") && nextTabs.length) {
+            const buttonElement = document.getElementById(`${nextTabs[0].id}-btn`) as HTMLButtonElement;
+            buttonElement.focus();
+            if (onChange) {
+                onChange(nextTabs[0].id);
+            }
+        }
+
+        if ((e.key === "ArrowLeft" || e.key === "ArrowUp") && previousTabs.length) {
+            const buttonElement = document.getElementById(
+                `${previousTabs[previousTabs.length - 1].id}-btn`,
+            ) as HTMLButtonElement;
```

Comment:
```suggestion
            const buttonElement = document.getElementById(
                `${previousTabs[previousTabs.length - 1].id}-btn`,
            );
            
            if (buttonElement) {
              ...
            } else {
            	throw ...
            }
```

Reply:
If it's better to throw an exception if the `buttonElement` is undefined, I changed it for a try/catch as this function already have too many if statement in my opinion. WDYT?